### DISCUSSION TITLE
단말 내에 설치된 test executable 파일 삭제

### DISCUSSION THREAD

**Discussion by YoungSeokHong** [euphony-io/euphony#98] [Ideas]
gtest가 끝나고 디바이스에 저장된 gtest실행에 관련된 필요없어진 파일들을 지우는 코드입니다.
원래 마지막에 시간 남으면 하려고 아끼고 있었는데 1주일 밖에 남지 않아서, 잠도 오지 않는 김에 급하게 만들어봤습니다!

### tests/CMakeLists.txt
```cmake
# Remove gtest executable files in device.
add_custom_command(TARGET ${TEST_EUPHONY} POST_BUILD
        COMMAND ${ADB} -s ${SERIAL} shell rm -r ${TARGET_TEST_LIB_DIR}
        )
```
원래는 이렇듯 테스트가 끝나고난 뒤에도 디바이스의 로컬 저장소에 불필요한 파일들이 남아있지만
![image](https://user-images.githubusercontent.com/80624082/136258741-2090e128-86eb-4bc0-b583-1b57dfb6450d.png)

위 코드를 CMakeLists의 if문 마지막에 추가해주면, 아래와 같이 테스트가 끝나고나서 testEuphony 디렉토리가 껍데기만 남고 불필요했던 파일들이 삭제됩니다.
![image](https://user-images.githubusercontent.com/80624082/136258901-f00b7f28-e67b-458a-be89-64339551ddde.png)

원래는 push받은 파일 4개를 하나씩 지워줄까 디렉토리를 통으로 날릴까 고민을 조금 했는데 후자가 코드도 깔끔하고 확장성도 좋을 것 같아 후자를 선택했습니다. 이부분에 대해서 피드백 부탁드립니다.

피드백 받고 바로 PR 올리겠습니다.

**Comment by kuro11pow2** [comment]
좋은 아이디어인 것 같습니다!
그런데 잔여 파일을 삭제하게 되면  shell에 직접 접속해서 디버그할 때 디바이스가 정상적으로 연결되었고 실행 파일이 업로드되었다는 사실을 확인하기 어려울 것 같습니다 ㅠㅠ 혹시 이를 알 수 있는 다른 방법이 있을까요?

**Reply by YoungSeokHong** [reply]
죄송합니다 제가 shell에 직접 접속해서 디버그를 하는 상황이 어떤 상황인지 정확히 모르겠어서ㅠㅠ 혹시 설명해주실 수 있나요??

**Reply by kuro11pow2** [reply]
대단한 건 아니고요 그냥 파일 수정 일자 확인하는 더블 체크 방식을 말씀 드린 거예요😄

**Reply by yeonns** [reply]
@kuro11pow2 님이 말씀하신 shell접속 후 testEuphony 바이너리 수정날짜 확인하는 부분 캡쳐해봤습니다 😄 빌드시에 매번 gtest가 돌아갔다면 testEuphony바이너리를 매번 adb push하기 때문에 해당 파일의 수정시간은 빌드한 시간과 일치하게 되어 더블체크할 때 쓰곤했습니다.
![image](https://user-images.githubusercontent.com/10149398/136390196-d8f11408-e412-4ab4-82ae-2c4051b681bd.png)

@YoungSeokHong @kuro11pow2 아무래도 잔여파일을 삭제하면 실행파일이 제대로 업로드되었는지 테스트는 수행되었는지 디버깅은 쉽지 않을것같습니다! 하지만 이제 gradle cache를 지워주는 패치도 올라왔고, 매번 빌드때마다 테스트를 수행하는게 보장되니 테스트 실행결과를
```.cxx/cmake/debug/ABI/android_gradle_build_stdout_targets``` 이 파일에서 확인할 수 있지 않을까? 라는 생각이 드네요 (stdout 파일도 이제 빌드시마다 계속 생성되서 항상 gtest 결과가 출력되더라구요!) 
```
C:/git/openup/euphony/euphony/build/intermediates/cmake/debug/obj/arm64-v8a/testEuphony: 1 file pushed, 0 skipped. 143.4 MB/s (11786368 bytes in 0.078s)
**** Gtest Success ****
```

**Reply by kuro11pow2** [reply]
아이고 맞습니다!! 제가 설명이 너무 부실했네요 ㅠㅠ 자세한 설명 감사드립니다👍 
말씀대로 gradle cache를 지워주는 패치가 적용되어서 매번 gtest 결과가 출력되니 이를 통해 확인할 수도 있겠네요 😄

**Reply by yeonns** [reply]
저번에 slack에서 나눴던 대화처럼 stdout 로그파일을 읽어서 android studio에 출력하는 gradle task를 만드는 걸 할수만있다면 확인하기가 더 편할거같다는 생각이 듭니다 ㅋㅋ!!

**Reply by kuro11pow2** [reply]
> 저번에 slack에서 나눴던 대화처럼 stdout 로그파일을 읽어서 android studio에 출력하는 gradle task를 만드는 걸 할수만있다면 확인하기가 더 편할거같다는 생각이 듭니다 ㅋㅋ!!

맞습니다!! 해야지 해야지 하면서 지금까지 미뤘는데 드디어 해야할 때가 온 것 같습니다 ㅋㅋㅋㅋ

**Comment by designe** [comment]
이거는 cmake 후속 미션으로 드리려고 했었는데 ㅋㅋㅋㅋㅋ
delete를 해주는 것이 맞을 것 같습니다.

이것도 PR로 올려주시죠!

그리고 추가적으로 매번 gtest의 결과를 
PC로 받아 .cxx 폴더 또는 .cxx/CMakeLists 폴더 내에 
출력 결과를 아래와 같은 이름의 파일로 내주는 것도
한번 해결해보고 싶으신 분 있으시다면 도전해주세요!!

성공시
- native-gtest-success-result-20211007112330.txt

실패시
- native-gtest-failed-result-20211007112330.txt

그리고 매회 유닛테스트 마다 테스트 결과는 삭제해주기

도전!! cmake와 adb를 보다 잘 이해할 수 있는 기회!!

**Reply by YoungSeokHong** [reply]
PR완료했습니다!

**Reply by YoungSeokHong** [reply]
이렇게 되면 매 빌드 마다 이전에 생성된 텍스트 파일을 지움과 동시에 현재시간의 새로운 텍스트 파일을 생성하는 것인가요??

**Reply by designe** [reply]
@YoungSeokHong 
@kuro11pow2 님이 gradle에서 결과를 출력하는 패치를 올리셔서 위의 추가적인 결과물을 따로 저장하지 않아도 될 것 같아요!
위 미션은 구현하지 않아도 될 것 같네요 :)

**Reply by YoungSeokHong** [reply]
앗 그러네요 ㅋㅋㅋ 한 발 늦었네요 ㅋㅋㅋㅋㅋㅋㅋ ㅠㅠ

**Reply by designe** [reply]
ㅋㅋㅋㅋㅋ 이번 그래들 패치는 예상치 못한 기여였습니다 ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ

**Comment by seohy02** [comment]
좋은 정보 감사합니다 !!!!

**Comment by Jsueeee** [comment]
👍 좋은 내용 감사합니다!

**Comment by judemin** [comment]
좋은 자료 감사합니다 :)

**Comment by dlwlstks96** [comment]
고생하셨습니다!!👍👍
